sdks/python/apache_beam/io/gcp/bigquery.py

```diff
@@ -607,7 +607,20 @@ def __init__(self, source, test_bigquery_client=None, use_legacy_sql=True,
 
   def __enter__(self):
     self.client = BigQueryWrapper(client=self.test_bigquery_client)
-    self.client.create_temporary_dataset(self.executing_project)
+
+    tr = self.source.table_reference
```

Comment:
Can we move this whole section inside the get_table_location function?
`def _get_source_table_location(self)`

and this `__enter__` would just look like `self.client.create_temporary_dataset(self.executing_project, self._get_source_table_location())`